index_release.go

```diff
@@ -19,6 +20,152 @@ type IndexRelease struct {
 	Version     string      `yaml:"version"`
 }
 
+// CompileReleases takes indexReleases and collected version bundles and
+// compiles canonicalized Releases from them.
+func CompileReleases(logger micrologger.Logger, indexReleases []IndexRelease, bundles []Bundle) ([]Release, error) {
+	releases := buildReleases(logger, indexReleases, bundles)
+
+	releases = deduplicateReleaseChangelog(releases)
+
+	return releases, nil
+}
+
+func buildReleases(logger micrologger.Logger, indexReleases []IndexRelease, bundles []Bundle) []Release {
+	bundleCache := make(map[string]Bundle)
+
+	// Create cache of bundles for quick lookup
+	for _, b := range bundles {
+		bundleCache[b.ID()] = b
+	}
+
+	var releases []Release
+
+	for _, ir := range indexReleases {
+		bundles, err := groupBundlesForIndexRelease(ir, bundleCache)
+		if IsBundleNotFound(err) {
+			logger.Log("level", "warning", "message", err.Error())
```

Comment:
Super nitpick, but I would like to have it like this to be more explicit. Below is also logging that we can align with this approach then. 

```
logger.Log("level", "warning", "message", "failed grouping version bundles for release", "stack", fmt.Sprintf("%#v", err))
```